docs/guides/workloads/deployment/index.md

```diff
@@ -81,578 +82,403 @@ apiVersion: apps/v1
 
-Now, create a `Repository` using this secret. Below is the YAML of `Repository` crd we are going to create,
+Now, create a `BackupStorage` using this secret. Below is the YAML of `BackupStorage` CR we are going to create,
 
 ```yaml
-apiVersion: stash.appscode.com/v1alpha1
-kind: Repository
+apiVersion: storage.kubestash.com/v1alpha1
+kind: BackupStorage
 metadata:
-  name: gcs-repo
+  name: gcs-storage
   namespace: demo
 spec:
-  backend:
+  storage:
+    provider: gcs
     gcs:
-      bucket: appscode-qa
-      prefix: /source/data/sample-deployment
-    storageSecretName: gcs-secret
+      bucket: kubestash-qa
+      prefix: demo
+      secretName: gcs-secret
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+  default: true
+  deletionPolicy: WipeOut
 ```
 
-Let's create the Repository we have shown above,
+Let's create the BackupStorage we have shown above,
 
 ```bash
-$ kubectl apply -f https://github.com/kubestash/docs/raw/{{< param "info.version" >}}/docs/guides/workloads/deployment/examples/repository.yaml
-repository.stash.appscode.com/gcs-repo created
+$ kubectl apply -f https://github.com/kubestash/docs/raw/{{< param "info.version" >}}/docs/guides/workloads/deployment/examples/backupstorage.yaml
+backupstorage.storage.kubestash.com/gcs-storage created
 ```
 
 Now, we are ready to backup our sample data into this backend.
 
-### Backup
-
-We have to create a `BackupConfiguration` crd targeting the `stash-demo` Deployment that we have deployed earlier. Stash will inject a sidecar container into the target. It will also create a `CronJob` to take periodic backup of `/source/data` directory of the target.
+**Create RetentionPolicy:**
 
-**Create BackupConfiguration:**
+Now, let's create a `RetentionPolicy` to specify how the old Snapshots should be cleaned up.
 
-Below is the YAML of the `BackupConfiguration` crd that we are going to create,
+Below is the YAML of the `RetentionPolicy` object that we are going to create,
 
 ```yaml
-apiVersion: stash.appscode.com/v1beta1
-kind: BackupConfiguration
+apiVersion: storage.kubestash.com/v1alpha1
+kind: RetentionPolicy
 metadata:
-  name: deployment-backup
+  name: demo-retention
   namespace: demo
 spec:
-  repository:
-    name: gcs-repo
-  schedule: "*/5 * * * *"
-  target:
-    ref:
-      apiVersion: apps/v1
-      kind: Deployment
-      name: stash-demo
-    volumeMounts:
-    - name: source-data
-      mountPath: /source/data
-    paths:
-    - /source/data
-  retentionPolicy:
-    name: 'keep-last-5'
-    keepLast: 5
-    prune: true
+  default: true
+  failedSnapshots:
+    last: 2
+  maxRetentionPeriod: 2mo
+  successfulSnapshots:
+    last: 5
+  usagePolicy:
+    allowedNamespaces:
+      from: All
 ```
+Notice the `spec.usagePolicy` that allows referencing the `RetentionPolicy` from all namespaces. To allow specific namespaces, we can configure it accordingly by following [RetentionPolicy usage policy](/docs/concepts/crds/retentionpolicy/index.md#retentionpolicy-spec).
```

Comment:
This part was not specified for the backupstorage. You can remove this to keep things consistent. 

> This change applies to all the relevant places